src/imitation/algorithms/adversarial/common.py

```diff
@@ -222,16 +279,22 @@ def __init__(
 
         self.venv_buffering = wrappers.BufferingWrapper(self.venv)
 
+        self.disc_trainer_callback = TrainDiscriminatorCallback(self)
```

Comment:
Why not define the `gen_callback` here like this:

```python
self.gen_callback: List[callbacks.BaseCallback] = [self.disc_trainer_callback]
```

and then just append it down in the else block?

And while you are at it rename it to use a plural because it is actually more than one callback. E.g. `self.gen_callbacks`.